knotx-core/src/main/java/io/knotx/proxy/AdapterProxy.java

```diff
@@ -31,5 +32,9 @@ static AdapterProxy createProxy(Vertx vertx, String address) {
     return new AdapterProxyVertxEBProxy(vertx, address);
   }
 
+  static AdapterProxy createProxyWithOptions(Vertx vertx, String address, DeliveryOptions deliveryOptions) {
```

Comment:
why you added the new createProxy method ? can it be just one createProxy() method consuming deliveryOptions ?
I see that you created Factory specifically for instantiation of Knot's, Adapters, etc. that uses createProxy method with or without deliveryOptions. But it's not needed at all. A ....VertxEBProxy implementation handles it for you - if you will set deliveryOptions to null it will not use it - you don't need to do any handling or factories.

Reply:
I'm just thinking out loud.
@marcinczeczko don't you agree that calling:
`AdapterProxy.createProxy(vertx, address, null)` looks odd?
Why the `null`? Should this interface force the user to pass `DeliveryOptions` parameter?
This is just to make the code look good, and skip in my opinion weird constructions like
`Factory.construct(param, null, null, param2)`.

Reply:
You are right that Vert.x handle nulls. But as a result we will have to add nulls in all tests. So we can have AdapterProxy.createProxy(vertx, address, null) in all tests.

Reply:
Yes, but I don't like overengineering by adding Factories to handle nulls, don't you think isn't too big for such small thing ?

Reply:
After the offline catchup with @marcinczeczko:
- let's not break compatibility and stay with 2 factory methods in `ProxyGen` helpers (like in this interface)
- classes like `XXXProxyFactory` are a bit overengineered, let's just construct Proxies directly where it is necessary
- `Optional` in config POJOs - suggestion here is to stick with Vert.x approach, and initialize not set properties with default values, e.g. in constructor:
```
...
deliveryOptions = config.containsKey("deliveryOptions") ? new DeliveryOptions(config.getJsonObject("deliveryOptions")) : new DeliveryOptions();
...
```

Reply:
Fixed.